File: app/backend/BakeryBackend/Models/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema; // Required for [Column]

namespace BakeryBackend.Models
{
    [Table("orders")] // Ensures it hits the correct table name
    public class Order
    {
        public Guid Id { get; set; }

        [Column("items")]
        public List<OrderItem> Items { get; set; } = new();

        [Column("subtotal")]
        public decimal Subtotal { get; set; }

        [Column("tax")]
        public decimal Tax { get; set; }

        [Column("total")]
        public decimal Total { get; set; }

        [Column("payment_type")]
        public string PaymentType { get; set; } = string.Empty;

        [Column("card_entry_method")]
        public string? CardEntryMethod { get; set; }

        [Column("cash_tendered")]
        public decimal? CashTendered { get; set; }

        [Column("change_given")]
        public decimal? ChangeGiven { get; set; }

        [Column("stripe_payment_id")]
        public string? StripePaymentId { get; set; }

        [Column("created_at")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;


        [Column("customer_id")]
        public string? CustomerId { get; set; }

        [Column("customer_name")]
        public string? CustomerName { get; set; }

        [Column("pickup_time")]
        public DateTime? PickupTime { get; set; }

        [Column("notes")]
        public string? Notes { get; set; }

        [Column("customer_email")]
        public string? CustomerEmail { get; set; }

        [Column("customer_phone")]
        public string? CustomerPhone { get; set; }

        [Column("fulfillment_type")]
        public string? FulfillmentType { get; set; }

        [Column("address")]
        public string? Address { get; set; }

        [Column("city")]
        public string? City { get; set; }

        [Column("state")]
        public string? State { get; set; }

        [Column("zip")]
        public string? Zip { get; set; }

        [Column("status")]
        public string Status { get; set; } = "pending"; // pending, paid, shipped, pickedUp, cancelled

        [Column("TrackingNumber")]
        public string? TrackingNumber { get; set; } // Optional for shipping

        [Column("FulfilledAt")]
        public DateTime? FulfilledAt { get; set; }

    }
}

File: app/types/order.ts
// app/manager/types.ts

export type ManagerOrder = {
  id: string;

  /* -------------------------------------------------------
     ACCOUNTING
  -------------------------------------------------------- */
  subtotal: number;
  tax: number;
  total: number;

  /* -------------------------------------------------------
     PAYMENT
  -------------------------------------------------------- */
  paymentType: string;              // "cash" | "card" | "debit"
  cardEntryMethod: string | null;   // "manual" | "terminal" | null
  cashTendered: number | null;
  changeGiven: number | null;
  stripePaymentId: string | null;

  /* -------------------------------------------------------
     CUSTOMER
  -------------------------------------------------------- */
  customerId: string | null;
  customerName: string | null;
  customerEmail: string | null;
  customerPhone: string | null;

  /* -------------------------------------------------------
     FULFILLMENT
  -------------------------------------------------------- */
  status: string;                   // "pending", "paid", "ready", "pickedUp", "completed", etc.
  fulfillmentType: string;          // "POS", "pickup", "shipping", "delivery"

  pickupTime: string | null;        // ISO timestamp
  fulfilledAt?: string;             // ISO timestamp

  // ⭐ Shipping-specific fields
  shippingCarrier?: string | null;  // "UPS", "USPS", "FedEx", etc.
  trackingNumber?: string | null;
  labelUrl?: string | null;         // URL to generated shipping label

  // ⭐ Delivery-specific fields
  courierName?: string | null;      // Uber/Lyft courier name
  courierPhone?: string | null;
  deliveryEta?: string | null;      // Estimated delivery time
  deliveredAt?: string | null;      // ISO timestamp

  /* -------------------------------------------------------
     ADDRESS (Shipping + Delivery)
  -------------------------------------------------------- */
  address: string | null;
  city: string | null;
  state: string | null;
  zip: string | null;

  /* -------------------------------------------------------
     NOTES
  -------------------------------------------------------- */
  notes: string | null;

  /* -------------------------------------------------------
     TIMESTAMPS
  -------------------------------------------------------- */
  createdAt: string;                // canonical backend timestamp

  /* -------------------------------------------------------
     ITEMS
  -------------------------------------------------------- */
  items: {
    product: {
      id: number;
      name: string;
      price: number;
      description: string;
      imageUrl: string;
      sortOrder: number;
    };
    quantity: number;
  }[];
};
